### DISCUSSION TITLE
Uncaught ReflectionException: Class "Deployer\Deployer" does not exist

### DISCUSSION THREAD

**Discussion by vesnindev** [deployphp/deployer#3103] [Help needed]
Hi. After update php cli from 7.4 to 8.1 I will get error:

```
PHP Fatal error: Uncaught ReflectionException: Class "Deployer\Deployer" does not exist in phar:///usr/local/bin/dep/bin/dep:119 Stack trace: #0 phar:///usr/local/bin/dep/bin/dep(119): ReflectionMethod->__construct() #1 /usr/local/bin/dep(4): require('...') #2 {main} thrown in phar:///usr/local/bin/dep/bin/dep on line 119
```

Also I try to reinstall deployer from https://deployer.org/download

but I also get error

Also If I try install Deployer via composer: composer require deployer/deployer --dev

I get error:

```
Problem 1 - deployer/deployer[v4.3.0, ..., 4.x-dev] require monolog/monolog ^1.21 -> found monolog/monolog[1.21.0, ..., 1.x-dev] but it conflicts with your root composer.json require (^2.0). - Root composer.json requires deployer/deployer ^4.3 -> satisfiable by deployer/deployer[v4.3.0, ..., 4.x-dev].

```
Deployer 7

Laravel 9, PHP 8.1, Minimal monolog version for Laravel 9 is "monolog/monolog": "^2.0"

**Comment by nicolastorgato** [comment]
Hi, I'm trying to install Deployer on a fresh laravel project and I receive the same error about v4.3.0...
I mean, how is that possible? It's the latest Laravel version...

Laravel 9.2, PHP 8.0.2

**Reply by MarkusPayne** [reply]
I was getting the 

PHP Fatal error: Uncaught ReflectionException: Class "Deployer\Deployer" does not exist in phar:///usr/local/bin/dep/bin/dep:119 

error after running "composer upgrade" on a project which bumped the version from from v7.0.0-rc.4 to v7.0.0-rc.8 and after some digging around I noticed the composer.lock file had changed the "bin" element fom "bin/dep" to "deployer.phar"

I downgraded by running "composer require deployer/deployer:v7.0.0-rc.4" which changed it back to "bin/dep" and it started woking again.

**Reply by nicolastorgato** [reply]
Wow now it works, thanks so much Markus! 

I'm just wondering... why did they change that from "bin/dep" to "deployer.phar"? Do we have to continue using the rc.4 and not the latest version?

**Reply by fvsadem** [reply]
I have the same problem with deployer v7.0.2 
its works when downgrading to v7.0.0-rc.4. 
Well, is that normal ?

Thanks

**Comment by kick-radiomeuh** [comment]
I have the same problem with deployer v7.1.2 & php8.2
in a desperate move, it works when downgrading to v7.0.0-rc.4 !

**Reply by kick-radiomeuh** [reply]
Now with php8.2 and Laravel9.*
The current deployer is v7.3.1
Always with this exact same bug
"Fatal error: Uncaught ReflectionException: Class "Deployer\Deployer" does not exist in phar:///usr/local/bin/dep/bin/dep:119"
and now unable to downgrade to 7.0.0-rc.4.

Is there a way to patch it somehow or
is it a planned move to be incompatible with Laravel ?
